nes-sources/src/LogicalSource.cpp

```diff
@@ -12,22 +12,23 @@
     limitations under the License.
 */
 
+#include <Sources/LogicalSource.hpp>
+
 #include <cstdint>
 #include <functional>
 #include <memory>
 #include <ostream>
 #include <string>
 #include <utility>
 #include <DataTypes/Schema.hpp>
-#include <Sources/LogicalSource.hpp>
 #include <fmt/format.h>
 
 namespace NES
 {
 
 
-LogicalSource::LogicalSource(std::string logicalSourceName, const std::shared_ptr<Schema>& schema)
-    : logicalSourceName(std::move(logicalSourceName)), schema(schema)
+LogicalSource::LogicalSource(std::string logicalSourceName, std::shared_ptr<const Schema>&& schema)
```

Comment:
Again, I can not think of any valid reason why you would pass a temporary to a shared pointer here. 
Scenario 1: You construct a temporary with `std::make_shared` when calling this ctor. In this case, it would make more sense to just create the object in here instead of passing from the outside.
Scenario 2: You have an lvalue and use `std::move` to move the shared pointer in this ctor. But shared pointers are for shared ownership, with move you give up the original value, removing the purpose of the shared pointer.